### ISSUE TITLE
Lint: force mandatory fields in `__construct`

### ISSUE THREAD

**Issue by homersimpsons** [phpstan/phpstan-doctrine#636]
## New lint description

PHPStan is really great to catch errors before they actually happen in production. But there is one error it does not catch (yet?):

```php
class User {
    #[ORM\Column(length: 180)]
    private string $email;

    public function getEmail(): string { return $this->email; }
    public function setEmail(string $email): string { $this->email = $email; }
}

$user = new User();
$manager->persist($user);
$manager->flush(); // SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'email' cannot be null
```

## Proposal

I believe it would be impossible, in a reasonable time, to check if every mandatory property is set before we call the `flush`.

But there may be a middle ground that should be achievable: enforce the developper to set all mandatory properties in constructor:

```php
class User {
    #[ORM\Column(length: 180)]
    private string $email;

    public function __construct(string $email) { $this->email = $email; }
}
```

This should be sufficient to handle most of the cases where this issue could arise.

### Lint report

Here are some example of how the lint would report issues:
```php
class User {
    #[ORM\Column(length: 180)]
    private string $email; // Mandatory property $email is not set in `__construct`
}
```
```php
class User {
    #[ORM\Column(length: 180)]
    private string $email;
    public function __construct() // Missing set of mandatory property $email
    {}
}
```

I do not know if it is possible to report on 2 different places. I think I find it clearer like this as the error is close to where the change should happen. But if that is not possible / too hard we can simply always add it on the property directly.

### Generated columns should not be counted as mandatory

The following example would not report an error as the mandatory column is generated.
```php
class User
{
    #[ORM\Id]
    #[ORM\Column(type: UuidType::NAME, unique: true)]
    #[ORM\GeneratedValue(strategy: 'CUSTOM')]
    #[ORM\CustomIdGenerator(class: 'doctrine.uuid_generator')]
    private Uuid $id;
}
```

### Promoted-property mandatory column

The following example would not report an error as the mandatory column is a promoted property:
```php
class User {
    public function __construct(
        #[ORM\Column(length: 180)]
        private string $email,
    ) {}
```

**Comment by ondrejmirtes** [comment]
Set `checkUninitializedProperties` to true. https://phpstan.org/config-reference#checkuninitializedproperties
